lib/pure/times.nim

```diff
@@ -1121,12 +1296,137 @@ proc evaluateStaticInterval(interval: TimeInterval): Duration =
     minutes = interval.minutes,
     hours = interval.hours)
 
+proc between*(start_dt, end_dt:DateTime, strict: bool = true): TimeInterval =
```

Comment:
Why do we need this when we have ``-`` which produces a ``Duration``?

Reply:
Cause here we have `TimeInterval` in result which is more convenient in some cases